rcl/src/rcl/timer.c

```diff
@@ -348,6 +422,8 @@ rcl_timer_exchange_period(const rcl_timer_t * timer, int64_t new_period, int64_t
   RCL_CHECK_ARGUMENT_FOR_NULL(timer, RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_ARGUMENT_FOR_NULL(old_period, RCL_RET_INVALID_ARGUMENT);
   *old_period = rcutils_atomic_exchange_uint64_t(&timer->impl->period, new_period);
+  rcutils_atomic_store(
+    &timer->impl->period_correction, timer->impl->options.initial_period_correction);
```

Comment:
should initial period jitter be reset here?

Reply:
I didn't because unlike feedback error, I don't expect jitter to change with the setpoint. But now that I think again about this, it might (e.g. due to the configured period becoming larger or smaller than the scheduler's time slot). We should reset it to a reasonable default.